### DISCUSSION TITLE
How to change legend size？

### DISCUSSION THREAD

**Discussion by k-user** [geopandas/geopandas#2711] [Q&A]
I have a drawing task, and I resized the image, but there is no way to resize legend using geodataframe.plot (). I see that the legend function in matplotlib has a BorderPad argument, but it does not apply in Geopandas. Can you help me? Thank you!

```python
fig, ax = plt.subplots(1, 1, figsize=(18, 25.76))
rain_df.plot(column='ID', ax=ax, alpha=0.6, cmap=colors.ListedColormap(rain_df['colors']), legend=True,
                 markersize=0.01,
                 categorical=True,
                 k=len(rain_df),
                 scheme='equal_interval',
                 legend_kwds={'labels': rain_df['level']}
                 )
```

**Comment by martinfleis** [comment]
You can do a bit with `legend_kwds=dict(prop={'size': 6})` but our legend handling is a known weak point so the markers do not actually scale as they should.

```py
df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))

df.plot('pop_est', scheme='quantiles', legend=True)
```
![Screenshot 2021-07-05 at 19 53 26](https://user-images.githubusercontent.com/36797143/124509918-abc6d900-ddca-11eb-9977-d93a0507a41b.png)

```py
df.plot('pop_est', scheme='quantiles', legend=True, legend_kwds=dict(prop={'size': 6}))
```


![Screenshot 2021-07-05 at 19 53 39](https://user-images.githubusercontent.com/36797143/124509933-b3867d80-ddca-11eb-8b03-ccfe4a2fbc07.png)
